week1/reposts/maverickchaser/Main.hs

```diff
@@ -0,0 +1,34 @@
+import Control.Monad
+import System.Random
+
+numberOfReposters       = 1000000
+numberOfFakeAccounts    = 100000
+numberOfSimulations     = 1000
+numberOfWinners         = 7
+
+
+generateAccounts :: Int -> Int -> StdGen -> [Int]
+generateAccounts nAccounts maxAccId gen =
+    take nAccounts $ randomRs (1, maxAccId) gen :: [Int]
+
+isWinner :: [Int] -> [Int] -> Bool
+isWinner fakeAccounts winners = any (==True) $ map (\acc -> acc `elem` winners) fakeAccounts
+
+simulate = do
+    gen <- newStdGen
+    let fakeAccounts = generateAccounts numberOfFakeAccounts numberOfReposters gen
+    gen <- newStdGen
+    let winners      = generateAccounts numberOfWinners numberOfReposters gen
```

Comment:
This might select the same person twice (which won't affect the result much, but still – can you fix that?).